File: PplPro/pplpro.client/src/app/models/employee.ts
import { Department } from "./department";
import { Role } from "./role";

export class Employee {
  employeeID: number = 0;
  name: string = "";
  position: string = "";
  salary: number = 0;
  departmentID: number = 0;
  department: Department  = new Department();
  roleID: number = 0;
  role: Role = new Role();

  constructor(init?: Partial<Employee>) {
    Object.assign(this, init);
  }
}

File: PplPro/PplPro.Server/Models/Employee/Employee.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PplPro.Server.Models
{
    public class Employee
    {
        [Key]
        public int EmployeeID { get; set; }

        [Required]
        [StringLength(100, MinimumLength = 2, ErrorMessage = "Name must be between 2 and 100 characters.")]
        public string Name { get; set; }

        [Required]
        [StringLength(50, ErrorMessage = "Position name cannot exceed 50 characters.")]
        public string Position { get; set; }

        [Required]
        [DataType(DataType.Currency)]
        public decimal Salary { get; set; }

        [Required]
        [ForeignKey("Department")]
        public int DepartmentID { get; set; }

        [Required]
        [ForeignKey("Role")]
        public int RoleID { get; set; }

        public Department Department { get; set; } // Add navigation property
        public Role Role { get; set; } // Add navigation property
    }
}
